.github/workflows/create-release-pr.yml

```diff
@@ -0,0 +1,240 @@
+        required: true
+        type: string
+      pre_release_version:
+        description: "Pre-Release version, e.g. 'beta.1'"
+        required: false
+        type: string
+
+env:
+  RELEASE_VERSION: ${{ inputs.release_version }}
+  PRE_RELEASE_VERSION: ${{ inputs.pre_release_version }}
+  RELEASE_BRANCH: release/${{ inputs.release_version }}
+
+jobs:
+  create-release-pr:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Set Release Version and Branch to Check Out
+        id: set-release
+        run: |
+          if [[ $RELEASE_VERSION =~ ^v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+            if [[ $PRE_RELEASE_VERSION =~ ^[a-z.0-9]+$ ]]; then
+              echo "release-tag: $RELEASE_VERSION-$PRE_RELEASE_VERSION"
+              echo "release-tag=$RELEASE_VERSION-$PRE_RELEASE_VERSION" >> $GITHUB_OUTPUT
+            elif [[ -n $PRE_RELEASE_VERSION ]]; then
+              echo "Input pre_release_version is not empty, but does not match the regex pattern ^[-a-z0-9]+$"
+              exit 1
+            else
+              echo "release-tag: $RELEASE_VERSION"
+              echo "release-tag=$RELEASE_VERSION" >> $GITHUB_OUTPUT
+            fi
+          else
+            echo "Version input doesn't match the regex pattern ^[0-9]+\.[0-9]+\.[0-9]+$"
+            exit 1
+          fi
+
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Create Release Branch if it does not exist
+        run: |
+          if ! git show-ref --verify --quiet "refs/remotes/origin/$RELEASE_BRANCH"; then
+            git checkout -b $RELEASE_BRANCH
+            git push --set-upstream origin $RELEASE_BRANCH
+          elif [[ $(git rev-parse --abbrev-ref HEAD) != "$RELEASE_BRANCH" ]]; then
+            echo "Current Branch: $(git rev-parse --abbrev-ref HEAD)"
+            echo "Release branch exists, make sure you're using the workflow from the release branch or delete the existing release branch."
+            exit 1
+          else
+            echo "Release branch exists and used as workflow ref."
+          fi
+
+      - name: Get Latest Release
+        id: get-release
+        run: |
+          if [[ -n $PRE_RELEASE_VERSION ]]; then
+            echo "Get the latest release"
+            tag=$(curl -L \
+                        --header "Accept: application/vnd.github.v3+json" \
+                        "https://api.github.com/repos/${{ github.repository }}/releases" | jq -r '.[0].tag_name')
+            echo "latest-tag=$tag" >> $GITHUB_OUTPUT
+          else
+            echo "Get the latest stable release"
+            tag=$(curl -L \
+                        --header "Accept: application/vnd.github.v3+json" \
+                        "https://api.github.com/repos/${{ github.repository }}/releases/latest" | jq -r '.tag_name')
+            echo "latest-tag=$tag" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Build Changelog
+        uses: mikepenz/release-changelog-builder-action@v3.7.2
```

Comment:
How confident are we using this guy's Github action plugin in our build process? Are there "official" alternatives, or might it be worth forking it into a repo of our own?

Reply:
Thank you for the comment!! The comment helped me take a step back and do some more research. Finally found the API I was looking for. I will update the other repos as well to use this API: https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#generate-release-notes-content-for-a-release